packages/arb-tx-aggregator/web3/eth.go

```diff
@@ -183,69 +141,84 @@ func buildCallMsg(args *CallTxArgs) ethereum.CallMsg {
 	if args.Data != nil {
 		data = *args.Data
 	}
-	return ethereum.CallMsg{
-		From:     from,
-		To:       args.To,
-		Gas:      gas,
-		GasPrice: gasPrice,
-		Value:    value,
-		Data:     data,
+
+	var dest arbcommon.Address
+	if args.To != nil {
+		dest = arbcommon.NewAddressFromEth(*args.To)
+	}
+	return from, message.ContractTransaction{
+		MaxGas:      new(big.Int).SetUint64(gas),
+		GasPriceBid: gasPrice,
+		DestAddress: dest,
+		Payment:     value,
+		Data:        data,
+	}
+}
+
+func (s *Server) executeCall(args *CallArgs) (*evm.TxResult, error) {
+	snap, err := s.getSnapshot(args.BlockNum)
+	if err != nil {
+		return nil, err
 	}
+	from, msg := buildCallMsg(args.CallArgs)
+	msg = s.srv.AdjustGas(msg)
+	return snap.Call(msg, from)
 }
 
-func (s *Server) Call(r *http.Request, args *CallArgs, reply *string) error {
-	ret, err := s.conn.CallContract(
-		r.Context(),
-		buildCallMsg(args.CallArgs),
-		big.NewInt(args.BlockNum.Int64()),
-	)
+func (s *Server) Call(_ *http.Request, args *CallArgs, reply *string) error {
+	res, err := s.executeCall(args)
 	if err != nil {
 		return err
 	}
-	*reply = hexutil.Encode(ret)
+	*reply = hexutil.Encode(res.ReturnData)
 	return nil
 }
 
-func (s *Server) EstimateGas(r *http.Request, args *CallTxArgs, reply *string) error {
-	ret, err := s.conn.EstimateGas(
-		r.Context(),
-		buildCallMsg(args),
-	)
+func (s *Server) EstimateGas(_ *http.Request, args *CallArgs, reply *string) error {
+	res, err := s.executeCall(args)
 	if err != nil {
 		return err
 	}
-	*reply = hexutil.EncodeUint64(ret)
+	*reply = hexutil.EncodeUint64(res.GasUsed.Uint64() + 1000000)
```

Comment:
Is it better to multiply by a factor like 1.1, rather than adding a constant?   Or maybe do both:  gasUsed*1.1 + constant?